plugins/snowflake/dbt/include/snowflake/macros/materializations/table.sql

```diff
@@ -49,51 +53,30 @@
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
-  -- build model
+  --build model
   {% call statement('main') -%}
-    {%- if non_destructive_mode -%}
-      {%- if old_relation is not none -%}
-        {{ create_table_as(create_as_temporary, intermediate_relation, sql) }}
-
-        {% set dest_columns = adapter.get_columns_in_relation(old_relation) %}
-        {% set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') %}
-
-        insert into {{ target_relation }} ({{ dest_cols_csv }}) (
-          select {{ dest_cols_csv }}
-          from {{ intermediate_relation.include(database=(not create_as_temporary), schema=(not create_as_temporary)) }}
-        );
-      {%- else -%}
-        {{ create_table_as(create_as_temporary, target_relation, sql) }}
-      {%- endif -%}
-    {%- else -%}
-      {{ create_table_as(create_as_temporary, intermediate_relation, sql) }}
-    {%- endif -%}
-  {%- endcall %}
-
-  -- cleanup
-  {% if non_destructive_mode -%}
-    -- noop
-  {%- else -%}
+  -- we can leverage Snowflake create or replace table here to achieve an atomic replace.
     {% if old_relation is not none %}
+      {# -- I'm preserving one of the old checks here for a view, and to make sure Snowflake doesn't
+      -- complain that we're running a replace table on a view. #}
       {% if old_relation.type == 'view' %}
-        {#-- This is the primary difference between Snowflake and Redshift. Renaming this view
-          -- would cause an error if the view has become invalid due to upstream schema changes #}
         {{ log("Dropping relation " ~ old_relation ~ " because it is a view and this model is a table.") }}
         {{ drop_relation_if_exists(old_relation) }}
-      {% else %}
-        {{ adapter.rename_relation(target_relation, backup_relation) }}
       {% endif %}
     {% endif %}
+    {{create_or_replace_table_as(target_relation, sql)}}
+  {%- endcall %}
 
-    {{ adapter.rename_relation(intermediate_relation, target_relation) }}
-  {%- endif %}
+  -- skiping all previous renames here since they are not needed in Snowflake
 
   {{ run_hooks(post_hooks, inside_transaction=True) }}
 
   -- `COMMIT` happens here
   {{ adapter.commit() }}
 
   -- finally, drop the existing/backup relation after the commit
+  {# -- TODO: Check with Drew wether this backup_relation gets used at all should this materialisation
```

Comment:
Here also, I didn't want to change the original logic but I am not sure whether that `backup_relation` serves much of a purpose, unless it is used by a later part of the run that may handle recovery using that backup or not. So I thought I'd bring also the attention of one of you to this bit.